docker-app/qfieldcloud/core/tests/test_jobs.py

```diff
@@ -0,0 +1,211 @@
+
+        # Push the QGIS project and datasource GPKG files
+        response = self.upload_file(self.p1, "bumblebees.gpkg", "bumblebees.gpkg")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        response = self.upload_file(
+            self.p1, "simple_bumblebees_wrong_localized.qgs", "project.qgs"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Create Package job and wait for it to finish
+        set_subscription(
+            self.u1,
+            "keep_10",
+            storage_mb=10,
+            is_premium=True,
+            is_external_db_supported=True,
+        )
+        package_job = PackageJob.objects.create(
+            type=Job.Type.PROCESS_PROJECTFILE,
+            project=self.p1,
+            created_by=self.u1,
+        )
+
+        self.assertEqual(package_job.status, Job.Status.PENDING)
+
+        wait_for_project_ok_status(self.p1)
+
+        package_job.refresh_from_db()
+
+        self.assertEqual(package_job.status, Job.Status.FINISHED)
+        self.assertIsNotNone(package_job.feedback)
+
+        # extract layer data from job
+        qgis_layers = package_job.get_feedback_step_data("qgis_layers_data")["returns"][
+            "layers_by_id"
+        ]
+        qfield_layers = package_job.get_feedback_step_data("qfield_layer_data")[
+            "returns"
+        ]["layers_by_id"]
+
+        # "valid" localized layer -> QFC considers it as invalid (QGIS)
+        self.assertLayerData(
+            qgis_layers["valid_localized_point_layer_id"],
+            is_valid=False,
+            is_localized=True,
+            error_code="localized_dataprovider",
+        )
+
+        # invalid localized layer (QGIS)
+        self.assertLayerData(
+            qgis_layers["invalid_localized_polygon_layer_id"],
+            is_valid=False,
+            is_localized=True,
+            error_code="localized_dataprovider",
+        )
+
+        # "valid" localized layer -> QFC considers it as invalid (QField)
+        self.assertLayerData(
+            qfield_layers["valid_localized_point_layer_id"],
+            is_valid=False,
+            is_localized=True,
+            error_code="localized_dataprovider",
+        )
+
+        # invalid localized layer (QField)
+        self.assertLayerData(
+            qfield_layers["invalid_localized_polygon_layer_id"],
+            is_valid=False,
+            is_localized=True,
+            error_code="localized_dataprovider",
+        )
+
+        # invalid layer is not present in qfield layers
+        self.assertNotIn("invalid_point_layer_id", qfield_layers)
+
+        # invalid layer (datasource does not exist) (QGIS)
+        self.assertLayerData(
+            qgis_layers["invalid_point_layer_id"],
+            is_valid=False,
+            is_localized=False,
+            error_code="invalid_dataprovider",
+        )
+
+    def assertLayerData(
```

Comment:
Can you move this to the start of the class, right before the first `test_` method?